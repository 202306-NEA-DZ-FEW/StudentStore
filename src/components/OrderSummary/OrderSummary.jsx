import React from "react";

import Button from "../Buttons/Button";

const OrderSummary = ({ cartItems, subtotal }) => {
    return (
        <div className='mt-6 h-full rounded-lg border bg-white p-6 shadow-md md:mt-0 md:w-[40%]'>
            <h2 className='text-2xl text-black font-bold text-center'>
                Order Summary
            </h2>
            <hr className='my-4' />
            <div className='flex justify-between'>
                <p className='text-lg text-black font-semibold'>Name</p>

                <p className='text-lg text-black font-semibold pr-3'>Price</p>
            </div>
            <hr className='my-2' />
            <div>
                {cartItems?.map((cartItem) => (
                    <div
                        key={cartItem.id}
                        className='mb-2 flex justify-between '
                    >
                        <p className='text-gray-700 font-medium w-2/3  truncate'>
                            {cartItem.title}
                        </p>
                        <p className='text-gray-700 font-semibold'>
                            {cartItem.price.toFixed(2)}$
                        </p>
                    </div>
                ))}
            </div>
            <hr className='my-4' />
            <div className='flex justify-between'>
                <p className='text-xl text-black font-bold'>Total</p>
                <div className=''>
                    <p className='mb-1 text-lg text-black font-bold'>
                        {subtotal?.toFixed(2)}$
                    </p>
                </div>
            </div>
            <Button className='mt-6 w-full  rounded-md  b font-medium text-blue-50 '>
                Check out
            </Button>
        </div>
    );
};

export default OrderSummary;
