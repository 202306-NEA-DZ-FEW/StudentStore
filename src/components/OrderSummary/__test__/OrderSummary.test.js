import renderer from "react-test-renderer";

import OrderSummary from "../OrderSummary";

it("renders correctly", () => {
    const tree = renderer.create(<OrderSummary />).toJSON();
    expect(tree).toMatchSnapshot();
});
