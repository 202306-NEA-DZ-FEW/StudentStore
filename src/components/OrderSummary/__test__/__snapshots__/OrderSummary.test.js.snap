// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mt-6 h-full rounded-lg border bg-white p-6 shadow-md md:mt-0 md:w-[40%]"
>
  <h2
    className="text-2xl text-black font-bold text-center"
  >
    Order Summary
  </h2>
  <hr
    className="my-4"
  />
  <div
    className="flex justify-between"
  >
    <p
      className="text-lg text-black font-semibold"
    >
      Name
    </p>
    <p
      className="text-lg text-black font-semibold pr-3"
    >
      Price
    </p>
  </div>
  <hr
    className="my-2"
  />
  <div />
  <hr
    className="my-4"
  />
  <div
    className="flex justify-between"
  >
    <p
      className="text-xl text-black font-bold"
    >
      Total
    </p>
    <div
      className=""
    >
      <p
        className="mb-1 text-lg text-black font-bold"
      >
        $
      </p>
    </div>
  </div>
  <button
    className="bg-[#FF8A57] border-2 border-[#FF8A57] py-2 px-14 shadow-sm text-md hover:bg-white hover:text-[#FF8A57] hover:border-2 hover:border-[#FF8A57] hover:duration-300 mt-6 w-full rounded-md b font-medium text-blue-50"
  >
    Check out
  </button>
</div>
`;
